experiment/simulation/js/main.js

```diff
@@ -1 +1,247 @@
-//Your JavaScript goes in here
+'use strict';
+
+document.addEventListener('DOMContentLoaded', function() {
+
+    const restartButton = document.getElementById('restart');
+    restartButton.addEventListener('click', function() { restart(); });
+
+    const playButton = document.getElementById('play');
+    playButton.addEventListener('click', function() { play(); });
+
+    const pauseButton = document.getElementById('pause');
+    pauseButton.addEventListener('click', function() { pause(); });
+
+    const slider = document.getElementById('speed');
+    const output = document.getElementById('demo_speed');
+    output.innerHTML = (slider.value) / 4;
+    slider.oninput = function() {
+        output.innerHTML = (this.value) / 4;
+        FPS = originalFPS * (output.innerHTML);
+        restart();
+    };
+
+
+    function restart() {
+        window.clearTimeout(tmHandle);
+        setAll();
+        play();
+    }
+
+    function play() {
+        if (flag1 === 0) {
```

Comment:
What do flag1 and flag2 represent? Is it possible to give them meaningful names?

Reply:
they are mark the end of some part of simulation.

Reply:
Yes so I asked if they can be given more meaningful names which capture the fact that they mark end of some part of simulation?
These were the first, basic things discussed on DASS.